src/event_loop.rs

```diff
@@ -0,0 +1,155 @@
+pub struct EventLoop<T> {
+    pub(crate) event_loop: platform_impl::EventLoop<T>,
+    pub(crate) _marker: ::std::marker::PhantomData<*mut ()> // Not Send nor Sync
+}
+
+/// Set by the user callback given to the `EventLoop::run` method.
+///
+/// Indicates the desired behavior of the event loop after [`Event::EventsCleared`][events_cleared]
+/// is emitted.
+///
+/// [events_cleared]: ../event/enum.Event.html#variant.EventsCleared
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ControlFlow {
+    /// When the current loop iteration finishes, suspend the thread until another event arrives.
+    Wait,
+    /// When the current loop iteration finishes, suspend the thread until either another event
+    /// arrives or the given time is reached.
+    WaitUntil(Instant),
+    /// When the current loop iteration finishes, immediately begin a new iteration regardless of
+    /// whether or not new events are available to process.
+    Poll,
+    /// Send a `LoopDestroyed` event and stop the event loop. Once set, `control_flow` cannot be
+    /// changed from `Exit`.
+    Exit
+}
+
+impl Default for ControlFlow {
+    #[inline(always)]
+    fn default() -> ControlFlow {
+        ControlFlow::Poll
+    }
+}
+
+impl EventLoop<()> {
+    /// Builds a new event loop with a `()` as the user event type.
+    pub fn new() -> EventLoop<()> {
+        EventLoop::<()>::new_user_event()
+    }
+}
+
+impl<T> EventLoop<T> {
+    /// Builds a new event loop.
+    ///
+    /// Usage will result in display backend initialisation, this can be controlled on linux
+    /// using an environment variable `WINIT_UNIX_BACKEND`. Legal values are `x11` and `wayland`.
+    /// If it is not set, winit will try to connect to a wayland connection, and if it fails will
+    /// fallback on x11. If this variable is set with any other value, winit will panic.
+    pub fn new_user_event() -> EventLoop<T> {
+        EventLoop {
+            event_loop: platform_impl::EventLoop::new(),
+            _marker: ::std::marker::PhantomData,
+        }
+    }
+
+    /// Returns the list of all the monitors available on the system.
+    ///
+    // Note: should be replaced with `-> impl Iterator` once stable.
+    #[inline]
+    pub fn get_available_monitors(&self) -> AvailableMonitorsIter {
+        let data = self.event_loop.get_available_monitors();
+        AvailableMonitorsIter{ data: data.into_iter() }
+    }
+
+    /// Returns the primary monitor of the system.
+    #[inline]
+    pub fn get_primary_monitor(&self) -> MonitorHandle {
+        MonitorHandle { inner: self.event_loop.get_primary_monitor() }
+    }
+
+    /// Hijacks the calling thread and initializes the `winit` event loop with the provided
+    /// closure. Since the closure is `'static`, it must be a `move` closure if it needs to
+    /// access any data from the calling context.
+    ///
+    /// Any values not passed to this function will *not* be dropped.
```

Comment:
I thnk we need to specify somewhere the behavior of the `&mut ControlFlow`, specifically regarding these points:

- What is its initial value?
- Is it reset to its initial value at the beginning of each event loop run?

Reply:
I've updated the docs to clarify this.